### DISCUSSION TITLE
Onedrive file sync "over 100%" and stuck on loop

### DISCUSSION THREAD

**Discussion by Nemesis533** [abraunegg/onedrive#2903] [Q&A] [answered]
Hello everyone, I'm having a very strange issue with the onedrive plugin in OMV and I can't seem to find an answer anywhere; here is a description of the issue:

- I can authenticate just fine
- Remote changes are downloaded just fine
- When it tries to upload changes, the first couple files go well, then it gets stuck into a "loop" where it keeps trying to upload the same files forever, going above 100% completion
- If I restart the service from CLI, it resumes these aborted uploads successfully, then moves on to other files and the same thign as above happens all over again.
- Right before it happenes, it loses connection to the MS server, retires to connect but fails

Running these versions:

ii  onedrive                        2.5.2-1+np1+1.1                      amd64        folder synchronization with OneDrive
ii  openmediavault-onedrive         7.1.0-1                              all          openmediavault OneDrive plugin

What has no worked thus far:
- reinstalling the plugin
- changing the target folder
- trying smaller files
- adding an explicit rule on the firewall to allow https traffic to the OMV machine (even if all other services relying on port 443 work just fine)
- the solution from this post https://github.com/abraunegg/onedrive/discussions/2884 did not work.
Thank you in advance to anyone that can help.

![Screenshot 2024-10-12 181252](https://github.com/user-attachments/assets/cf7ef1d0-801f-4318-891e-c07e4166a4cb)

![Screenshot 2024-10-12 155708](https://github.com/user-attachments/assets/3dfe2d53-bfea-4525-968b-d39a732cf72d)

What is your OneDrive Application Configuration
```
Application version                          = onedrive v2.5.2-1+np1+1.1
Compiled with                                = LDC 2100
User Application Config path                 = /root/.config/onedrive
System Application Config path               = /etc/onedrive
Applicable Application 'config' location     = /root/.config/onedrive/config
Configuration file found in config location  = true - using 'config' file values to override application defaults
Applicable 'sync_list' location              = /root/.config/onedrive/sync_list
Applicable 'items.sqlite3' location          = /root/.config/onedrive/items.sqlite3
Config option 'drive_id'                     =
Config option 'sync_dir'                     = ~/OneDrive
Config option 'enable_logging'               = false
Config option 'log_dir'                      = /var/log/onedrive
Config option 'disable_notifications'        = false
Config option 'skip_dir'                     =
Config option 'skip_dir_strict_match'        = false
Config option 'skip_file'                    = ~*|.~*|*.tmp|*.swp|*.partial
Config option 'skip_dotfiles'                = false
Config option 'skip_symlinks'                = false
Config option 'monitor_interval'             = 300
Config option 'monitor_log_frequency'        = 12
Config option 'monitor_fullscan_frequency'   = 12
Config option 'read_only_auth_scope'         = false
Config option 'dry_run'                      = false
Config option 'upload_only'                  = false
Config option 'download_only'                = false
Config option 'local_first'                  = false
Config option 'check_nosync'                 = false
Config option 'check_nomount'                = false
Config option 'resync'                       = false
Config option 'resync_auth'                  = false
Config option 'cleanup_local_files'          = false
Config option 'classify_as_big_delete'       = 1000
Config option 'disable_upload_validation'    = false
Config option 'disable_download_validation'  = false
Config option 'bypass_data_preservation'     = false
Config option 'no_remote_delete'             = false
Config option 'remove_source_files'          = false
Config option 'sync_dir_permissions'         = 700
Config option 'sync_file_permissions'        = 600
Config option 'space_reservation'            = 52428800
Config option 'application_id'               = d50ca740-c83f-4d1b-b616-12c519384f0c
Config option 'azure_ad_endpoint'            =
Config option 'azure_tenant_id'              =
Config option 'user_agent'                   = ISV|abraunegg|OneDrive Client for Linux/v2.5.2-1+np1+1.1
Config option 'force_http_11'                = false
Config option 'debug_https'                  = false
Config option 'rate_limit'                   = 0
Config option 'operation_timeout'            = 3600
Config option 'dns_timeout'                  = 60
Config option 'connect_timeout'              = 10
Config option 'data_timeout'                 = 60
Config option 'ip_protocol_version'          = 0
Config option 'threads'                      = 8
Environment var 'XDG_RUNTIME_DIR'            = false
Environment var 'DBUS_SESSION_BUS_ADDRESS'   = false
Config option 'notify_file_actions'          = false
```

**Comment by abraunegg** [comment, marked as answer]
@Nemesis533 
You have a 'curl' issue.

You must get OMV to provide you with a newer version of curl.

Your other options (which you have not done), it to follow the [Basic Troubleshooting Steps](https://github.com/abraunegg/onedrive?tab=readme-ov-file#basic-troubleshooting-steps) to downgrade various items .. but good luck in getting that done due to the OMV developer and the plugin not supporting this sort of user configuration.

**Actions|Options for you**
* Get OMV to provide a better version of curl
* Upgrade curl yourself
* Stop using the OMV 'onedrive' plugin and manage the client yourself under OMV
* Configure the client as per Basic Troubleshooting Steps

**Reply by Nemesis533** [reply]
@abraunegg thank you for the quick response! My bad for not including that info as well - I did look into the documentation, but to no avail - no curl update attempt or configuration change would help in any way with the omv-onedrive plugin. The solution was to install it myself and to set ip_protocol_version = "1" as well as force HTTP 1.1. Things appear to be up/downloading correctly now.

**Reply by abraunegg** [reply]
Please note that OMV manual configuration *dos not stick* ... the OMV developer does things in odd ways with that platform.

Perhaps go to the OMV forums and provide this feedback that:
* They need to provide a newer curl version
* Provide a capability to add options needed to make their included *broken* curl version work

Or just dont use such platform ... but thats a different discussion

**Reply by Nemesis533** [reply]
@abraunegg will definitely provide feedback to OMV about this issue; my main reason for going OMV is that TrueNas doesn't really support RAID5, and for general use the WebUI is actually quite useful (especially in managing users and shared folder access).

And thx for the warning - I'll come up with something that can ensure that the configuration is restored as needed.

**Reply by abraunegg** [reply]
FYI - TrueNAS supports ZFS which is way better than RAID5 for your data ..... I use ZFS for nearly everything - have done for 20+ years. 

On systems where I used to rely on RAID5 - hardware or software versions - I have had silent data corruption and data loss.

If you care about your data - use ZFS. Do not use anything else.

**Reply by Nemesis533** [reply]
True, I had not consider bit rot ... Thx for the advice and help! :)
